esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -2555,22 +2555,15 @@ bool CWsDfuEx::checkDescription(const char *description, const char *description
     if (descriptionFilter[len - 1] == '*')
         filterType += 2;
 
-    char descFilter[256];
+    StringBuffer descFilter;
     if (filterType < 1)
-        strcpy(descFilter, descriptionFilter);
+        descFilter.append(descriptionFilter);
     else if (filterType < 2)
-        strcpy(descFilter, descriptionFilter+1);
+        descFilter.append(descriptionFilter+1);
     else if (filterType < 3)
-    {
-        strncpy(descFilter, descriptionFilter, len - 1);
-        descFilter[len -2] = 0;
-    }
+        descFilter.append(len-1, descriptionFilter);
     else
-    {
-        strncpy(descFilter, descriptionFilter+1, len - 2);
-        descFilter[len -3] = 0;
-    }
-
```

Comment:
I think the last two cases previously incorrectly removed an extra character from the description.